blockchain/check-backup.js

```diff
@@ -67,23 +66,35 @@ stream.on("finish", async () => {
 
     const hash = await createHash(extractPath);
     const isValidMetadataFile = config.DirectoryHash === hash;
+
+    //Check for major version compatibility
+    const compatibleVersions =
+      config.hasOwnProperty("Version") &&
+      config.Version.split(".")[0] === version.split(".")[0];
+
     if (isValidMetadataFile) {
-      console.log(`The provided backup file is valid\n`);
-      console.log(`No updated backup is created`);
+      console.log("The provided backup file is valid\n");
+      console.log("No updated backup is created");
```

Comment:
Try not to use else statements when possible: Instead, check with if and do a process.exit(1)